task_DEV-3/Program.cs

```diff
@@ -0,0 +1,120 @@
+﻿using System;
+using System.Linq;
+
+namespace task_DEV_3
+{
+    // A program that determines whether the introduced nonnegative integer 
+    // is a member of the Fibonacci sequence. 
+    // Entered number size is limited to 254.
+    class Program
+    {
+        static void Main(string[] args)
+        {
+            Console.WriteLine("Enter a non-negative integer number.");
+            string inputValue = Console.ReadLine();
+
+            // Representation of the entered string value of the number 
+            // as an array of integers of size inputNumberPartSize each.
+            // Check the format of the entered number.
+            int inputNumberPartSize = int.MaxValue.ToString().Length - 1;
+            int inputNumberPartsAmount = inputValue.Length /
+                (inputNumberPartSize) + 1;
+            int[] inputNumberAsArray = new int[inputNumberPartsAmount];
+
+            try
+            {
+                for (int i = 0; i < inputNumberPartsAmount - 1; i++)
+                {
+                    inputNumberAsArray[i] = Convert.ToInt32(inputValue.
+                        Substring((inputNumberPartsAmount - 2 - i) * (inputNumberPartSize) +
+                        inputValue.Length % (inputNumberPartSize),
+                        inputNumberPartSize));
+                }
+                inputNumberAsArray[inputNumberPartsAmount - 1] = Convert.ToInt32(inputValue.
+                    Substring(0, inputValue.Length % (inputNumberPartSize)));
+            }
+            catch (FormatException)
+            {
+                Console.WriteLine("Sorry, the entered number is incorrect! Exiting process.");
+                Environment.Exit(-1);
+            }
+
+            // Iterative calculation of the members of the Fibonacci row
+            // and comparison with the introduced.
+            // Two previously calculated numbers are stored in previousRowMembers.
+
+            int[,] previousRowMembers = new int[2, inputNumberPartsAmount];
+            previousRowMembers[0, 0] = 0;
+            previousRowMembers[1, 0] = 1;
+            int[] currentRowMember = new int[inputNumberPartsAmount];
+            currentRowMember[0] = 1;
+
+            bool completeCalculation = false;
+            while (true)

```

Comment:
Too complicated unclear logic, simplify, Fibonacci number has to be calculated in a separate method